### PR TITLE
Update documentation and default build behavior for public repository

### PR DESCRIPTION
**Pull request by mcgibbon**
This PR performs updates to the documentation and changes the default build behavior (to use intermediate images) in response to the repository becoming public.

Changes:
- removed outdated CircleCI configuration README as it no longer reflects the way authentication is done on CircleCI
- removed pseudorandom string from documentation URL
- default to build from intermediate images, as this is our normal method of operation and the approach we would recommend to external users
- removed header from README for better integration into the sphinx documentation
- better linking with fv3gfs-util documentation using sphinx.ext.intersphinx
- removed noindex, nofollow headers from documentation
- re-labelled FV3GFS to fv3gfs-wrapper in documentation to reflect new project name
- added overall description of the project in documentation index, including flow chart image
- fixed linting errors in rst files, mostly to do with header styles

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -51,7 +58,7 @@
 master_doc = "index"
 
 # General information about the project.
-project = u"FV3GFS"
+project = u"fv3gfs-wrapper"
 copyright = u"2019, Vulcan Technologies, LLC"
```
Comment (eddie-c-davis) [inline docs/conf.py:62]: Does the copyright year need to be updated?

File: docs/usage.rst [hunk 2]
```diff
@@ -27,9 +24,8 @@ segmentation fault. The docker image is set up to use this setting by default in
 Running the model
 -----------------
 
-Basic model operation can be seen in this short, self-explanatory example::
+Basic model operation can be seen in this short, mostly self-explanatory example::
 
-    from mpi4py import MPI
```
Comment (eddie-c-davis) [inline docs/usage.rst:32]: Just curious where the `mpi4py` import moved, into the wrapper?

**Review by eddie-c-davis — APPROVED** [review]
Very nicely done. No obvious typos, just have a couple of small questions.

File: docs/usage.rst (hunk 2 above)
Comment (mcgibbon) [reply docs/usage.rst:32]: Yes, it was also in the wrapper before.

File: docs/conf.py (hunk 1 above)
Comment (mcgibbon) [reply docs/conf.py:62]: Good point.

Comment (mcgibbon) [inline docs/conf.py:62]:
```suggestion
copyright = u"2019-2020, Vulcan Technologies, LLC"
```

**Review by nbren12 — APPROVED** [review]
All good. Thanks for the docs improvements.

**Comment by nbren12** [comment]
I'll merge this since JM is out of office today.

**Pull request merged**